src/foam/nanos/auth/ruler/PreventDuplicateEmailAction.js

```diff
@@ -67,5 +84,39 @@ foam.CLASS({
         }
       `
     }
+  ],
+
+  static: [
+    {
+      name: 'spidGrantsDuplicateEmailPermission',
+      type: 'Boolean',
+      documentation: `
+      Common function for checking if the given SPID allows
+      duplicate emails.
+      `,
+      args: 'foam.core.X x, String spid',
+      javaCode: `
+      // this is a hack that prevents a crash in CRUNCH.
+      // if this isn't here, NPEs will happen.
+      if ( x.get("crunchService") == null ) {
```

Comment:
do you still need to check here?

Reply:
it needs to be checked here because there's no guarantee that X will always contain the crunchService. continuing without a crunchService present = crash. i'll rephrase the comment.